### PR TITLE
test_setup: add several functions to handle hugepages

### PR DESCRIPTION
**Pull request by czhang0**
Add 3 functions to achieve following:
1. set kernel hugepages
2. get the number of certain hugepages totally allocated by kernel
3. get the number of certain hugepages for given NUMA node

Signed-off-by: cuzhang <cuzhang@redhat.com>

### CODE REVIEW

**Comment by czhang0** [comment]
@luckyh ,

below is relevant commands used by similar fuctions in avocado.utils.memory 
grep Hugepagesize /proc/meminfo --> get default hugepage size
/sbin/sysctl vm.nr_hugepages --> get number of default hugepage size

we can get that the fuctions in avocado.utils.memory is only to handle default hugepages size, such as 2048K, so it can't be used to set/get for other huge pages, such as 1G.

So provide more common function to handle each hugepages size.

**Comment by luckyh** [comment]
OK, thanks for your explanation, then I would suggest extending the original interfaces by adding the optional argument `pagesize` instead of creating the new ones.

File: virttest/test_setup.py [hunk 1]
```diff
@@ -451,6 +451,63 @@ def get_multi_supported_hugepage_size(self):
...
+
+        if not isinstance(int(pagenum), type(0)):
+            raise ValueError("pagenum should be integer, while got "
+                             "%s" % pagenum)
+        if not os.path.isdir(pagedir) or not os.path.isfile(pagefile):
+            raise ValueError("Root hugepages control sysfs directory %s or"
+                             " file %s did not exist" % (pagedir, pagefile))
+        process.system("echo %s > %s" % (pagenum, pagefile), shell=True)
+
+    def get_free_hugepages(self, node, pagesize):
```
Comment (balamuruhans) [inline virttest/test_setup.py]: test_setup already have this API as `get_node_num_huge_pages(self, node, pagesize)`

File: virttest/test_setup.py [hunk 2]
```diff
@@ -451,6 +451,63 @@ def get_multi_supported_hugepage_size(self):
             raise ValueError("Root hugepage control sysfs directory %s did not"
                              " exist" % self.pool_path)
 
+    def get_kernel_hugepages(self, pagesize):
```
Comment (balamuruhans) [inline virttest/test_setup.py]: please check HugePageConfig() in test_setup and it can be enhanced if required

**Comment by sathnaga** [comment]
@czhang0 Based on above comments, I closing this PR for now, pls feel free to open it to
based on the requirement to implement a new API/reuse and enhance the old ones as suggested.
Thanks in advance.

**Comment by kylazhang** [comment]
Please keep this pr open as we need it for current testing, and I will update later according to the comments

**Pull request closed without merging**